src/backend/marsha/core/api/video.py

```diff
@@ -651,6 +664,60 @@ def get_queryset(self):
         # public context, we can't read any liveregistration
         return LiveRegistration.objects.none()
 
+    @action(detail=False, methods=["post"])
+    # pylint: disable=unused-argument
+    def push_attendance(self, request, pk=None):
+        """View handling pushing new attendance"""
+        serializer = serializers.LiveAttendanceSerializer(data=request.data)
+        if not serializer.is_valid():
+            return Response({"detail": "Invalid request."}, status=400)
+
+        user = self.request.user
+        video = get_object_or_404(Video, pk=user.id)
+
+        if self.is_lti_token():
+            token_user = user.token.payload.get("user")
+            consumer_site = get_object_or_404(
+                ConsumerSite, pk=user.token.payload["consumer_site"]
+            )
+            liveregistration, _ = LiveRegistration.objects.get_or_create(
+                consumer_site=consumer_site,
+                lti_id=user.token.payload.get("context_id"),
+                lti_user_id=token_user.get("id"),
+                video=video,
+            )
+            # Update liveregistration email only if it's defined in the token user
+            if token_user.get("email"):
+                liveregistration.email = token_user.get("email")
+
+            # Update liveregistration username only it's defined in the token user
+            if token_user.get("username"):
+                liveregistration.username = token_user.get("username")
+
+            # update or add live_attendance information
+            if liveregistration.live_attendance:
+                liveregistration.live_attendance = {
+                    **serializer.data["live_attendance"],
+                    **liveregistration.live_attendance,
+                }
```

Comment:
```suggestion
                liveregistration.live_attendance = serializer.data["live_attendance"] | liveregistration.live_attendance
```

discovered today https://www.python.org/dev/peps/pep-0584/

Reply:
thanks for sharing!